### ISSUE TITLE
Long LineString "jumps" when zooming in/out OR when dragging.

### ISSUE THREAD

**Issue by mjtworks** [visgl/deck.gl#6154]
#### Description

A long LineString "jumps" when we zoom in/out OR when Google Map is dragged a bit at certain zoom levels.
The distance of these jumps are most at the center of the LineString, and tapers down as we check towards the vertices.

#### Expected Behavior

The path taken by the LineString should be same regardless of the zoom level.

#### Repro Steps

  1. Create a basic deck.gl example that uses Google Maps.
  2. Choose 2 points that make up a long LineString on the map.
  3. Zoom in to level 12 keeping the center of the screen towards the center of the LineString.
  4. Zoom in to level 13. The LineString may disappear. If not, try dragging the map a bit.
  5. Zoom in to level 14. The LineString will disappear - the LineString can be seen again by moving East (for the given data set).

  - A Codepen that reproduces the behavior: [deck.gl and Google Maps example](https://codepen.io/mjtworks/pen/abwNJVW)
  - A sample dataset:
```js
  {
    features: [
      {
        geometry: { coordinates: [25.124497, 70.620571], type: "Point" },
        id: "e01",
        properties: { title: "East01" },
        type: "Feature"
      },
      {
        geometry: { coordinates: [24.985005, 70.635828], type: "Point" },
        id: "w01",
        properties: { title: "West01" },
        type: "Feature"
      },
      {
        geometry: { coordinates: [24.400711, 70.120828], type: "Point" },
        id: "e02",
        properties: { title: "East02" },
        type: "Feature"
      },
      {
        geometry: { coordinates: [24.138953, 70.142996], type: "Point" },
        id: "w02",
        properties: { title: "West02" },
        type: "Feature"
      },
      {
        geometry: {
          coordinates: [
            [8.020553, 58.160913],
            [25.826148, 71.113638]
          ],
          type: "LineString"
        },
        id: "line01",
        properties: { title: "LongLine01" },
        type: "Feature"
      }
    ],
    type: "FeatureCollection"
  }
```

#### Environment
 - Framework Version: deck.gl 8.5.6
 - Browser Version:
   - Firefox Version 91.0.2 (64-bit)
   - Chromium Version 92.0.4515.159 (Official Build) snap (64-bit)
   - Microsoft Edge Version 92.0.902.84 (Official build) (64-bit)
   - Google Chrome Version 93.0.4577.63 (Official Build) (64-bit)
   - Firefox Developer Version 92.0b9 (32-bit)
 - OS:
   - Ubuntu 21.04
   - Windows 10 Pro for Workstations / Version 21H1 / OS Build: 19043.1165 / Windows Feature Experience Pack 120.2212.3530.0

#### Logs

No errors are reported in the console.

**Comment by Pessimistress** [comment]
This is fixed in v8.5.8.

**Comment by mjtworks** [comment]
Many thanks @Pessimistress
